### ISSUE TITLE
Filters not reapplied after failures (web service temporarily unavailable)

### ISSUE THREAD

**Issue by Aeyoun** [getnikola/nikola#2169]
Scanning posts...............done!
    .  render_posts:timeline_changes
    .  render_archive:output/2015/index.html
    .  copy_files:output/assets/css/article.css
    [2015-11-08T18:15:30Z] ERROR: Nikola: can't use cssminifier.com: HTTPConnectionPool(host='webcache.net.daniel.priv.no', port=3128): Max retries exceeded with url: http://cssminifier.com/raw (Caused by ProxyError('Cannot connect to proxy.', gaierror(-5, 'No address associated with hostname')))
    [d@Orlin blog]$ export http_proxy=""
    [d@Orlin blog]$ nikola build
    Scanning posts...............done!

**Comment by Kwpolska** [comment]
We would need to have some way to mark the output file as outdated in that case.

**Comment by ralsina** [comment]
AFAIK doit doesn´t support that (which I would like to have for other
things) so we would have to add it somehow.

**Comment by Kwpolska** [comment]
Would this work?
```py
task['uptodate'].append(filter_success)

**Comment by ralsina** [comment]
Ok, I researched this a little.

The main problem is that filters are resilient: they never actually fail. For example, cssminify just returns the data unchanged.

So, one way to solve this is that filters should "somehow" report their failure, for example by letting exceptions leak out, and then that should be handled somehow, and the task marked as failed by that last action (unlessLink function in utils.py).

I did a little POC https://github.com/getnikola/nikola/pull/2245 (enable cssminify filter to see how it fails)

OTOH this is maybe not even useful. Currently you get error messages and things move along. With this kind of handling, you end up with a failing build. Which one is better? I am not sure.
